### PR TITLE
Make PIR available in the result of liftCode

### PR DESCRIPTION
**Pull request by zliu41**
This enables us to see the PIR of many test cases, which is useful for debugging.

### CODE REVIEW

File: plutus-tx/src/PlutusTx/Lift.hs [hunk 1]
```diff
@@ -85,10 +85,10 @@ safeLift v x = do
           -- that takes all the compilation options and everything.
           & set (ccOpts . coDatatypes . dcoStyle) (if v >= PLC.plcVersion110 then SumsOfProducts else ScottEncoding)
         ucOpts = PLC.defaultCompilationOpts & PLC.coSimplifyOpts . UPLC.soMaxSimplifierIterations .~ 0
-    plc <- flip runReaderT ccConfig $ compileProgram (Program () v lifted)
+    plc <- flip runReaderT ccConfig $ compileProgram (Program () v pir)
     uplc <- flip runReaderT ucOpts $ PLC.compileProgram plc
     (UPLC.Program _ _ db) <- traverseOf UPLC.progTerm UPLC.deBruijnTerm uplc
```
Comment (effectfully) [inline plutus-tx/src/PlutusTx/Lift.hs:90]: What's `db`?

File: plutus-tx/src/PlutusTx/Lift.hs [hunk 2]
```diff
@@ -101,8 +101,8 @@ safeLiftProgram
        , PrettyPrintable uni fun
        , Default (PLC.CostingPart uni fun)
        )
-    => PLC.Version -> a -> m (UPLC.Program UPLC.NamedDeBruijn uni fun ())
-safeLiftProgram v x = UPLC.Program () v <$> safeLift v x
+    => PLC.Version -> a -> m (PIR.Program PLC.TyName PLC.Name uni fun (), UPLC.Program UPLC.NamedDeBruijn uni fun ())
+safeLiftProgram v x = (PIR.Program () v *** UPLC.Program () v) <$> safeLift v x
```
Comment (effectfully) [inline plutus-tx/src/PlutusTx/Lift.hs:105]: I think `bimap` is a bit more standard, but I'm obviously nitpicking.

**Review by effectfully — APPROVED** [review]

File: plutus-tx/src/PlutusTx/Lift.hs (hunk 1 above)
Comment (zliu41) [reply plutus-tx/src/PlutusTx/Lift.hs:90]: I think it means De Bruijn term.

**Comment by michaelpj** [comment]
I don't really like this, I think it makes the interface much less usable. We could just have these functions return PIR programs (as the underlying `Lift` class does), and only compile them to UPLC when we construct `CompiledCode`?

**Comment by zliu41** [comment]
> We could just have these functions return PIR programs (as the underlying Lift class does)

That would require bigger changes to the test code, although the test code doesn't necessarily represent how usable the interface is. How about providing separate functions returning PIR and UPLC?

**Comment by effectfully** [comment]
> I don't really like this, I think it makes the interface much less usable. We could just have these functions return PIR programs (as the underlying `Lift` class does), and only compile them to `UPLC` when we construct `CompiledCode`?

Given this, I think I should clarify that my approvals for pull requests related to Plutus Tx, especially the compiler, shouldn't be treated too seriously. I don't really know anything about the Plutus Tx compiler.

**Pull request merged**